pkg/admin/prerun/prerun.go

```diff
@@ -76,20 +90,84 @@ func Perform() error {
 		return nil
 	}
 
-	if err := dataManager.Backup(newBackupName); err != nil {
+	if err := pr.dataManager.Backup(newBackupName); err != nil {
 		return err
 	}
 
-	removeOldBackups(dataManager, backupsForDeployment)
+	pr.removeOldBackups(backupsForDeployment)
 
 	return nil
 }
 
+func (pr *PreRun) restore() error {
+	// TODO: Check if containers are already running (i.e. microshift.service was restarted)?
```

Comment:
Situation when system was unhealthy, so it restored the backup, but then crashed (or exited due to IP change or smth else)  and was restarted before red/green script had chance to update `health.json`: it would restore again.

At this point some Pods might already run, so should we protect against restoring in such case? Or maybe it's not a problem.

Reply:
As a first iteration, let's assume it's safe to restore from the same backup multiple times.